### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by dz333n**
[Build status](https://ci.appveyor.com/project/feel-the-dz3n/reactosdbg/branch/improvements-time)

Major changes:
 - Upgrade projects to .NET 4.0;
 - Update DockPanel Suite (use latest code).

Minor, but still important changes:
 - Edit and improve some controls;
 - Use light VS2015 theme for DockPanel;
 - Refactor some code;
 - Add "Source Code" in help menu;
 - Use JIRA instead of Bugzilla.

How program looks now:
image

image

### CODE REVIEW

File: DebugProtocol/DebugConnection.cs [hunk 1]
```diff
@@ -339,8 +339,7 @@ void ThreadListEvent(object sender, ThreadListEventArgs args)
                 mCurrentProcess = mNewCurrentProcess;
                 mCurrentThread = mNewCurrentThread;
                 mProcesses = mAccumulateProcesses;
-                if (DebugProcessThreadChangeEvent != null)
-                    DebugProcessThreadChangeEvent(this, new DebugProcessThreadChangeEventArgs(mProcesses));
+                DebugProcessThreadChangeEvent?.Invoke(this, new DebugProcessThreadChangeEventArgs(mProcesses));
```
Comment (learn-more) [inline DebugProtocol/DebugConnection.cs]: If you are doing crap like this, please keep it separate from logic changes, it makes reviewing really annoying.

File: README.md [hunk 2]
```diff
@@ -9,4 +9,4 @@
 
 ## Building
 
-Use Visual Studio 2010+ to build this program. 
+Use Visual Studio 2017 or later to build this program.
```
Comment (learn-more) [inline README.md]: Why 2017?
What is wrong with 2013 or 2015?

File: WinFormsUI/Docking/DockContent.cs [hunk 3]
```diff
@@ -295,10 +295,8 @@ private void DockHandler_DockStateChanged(object sender, EventArgs e)
 		}
 		protected virtual void OnDockStateChanged(EventArgs e)
 		{
-			EventHandler handler = (EventHandler)Events[DockStateChangedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler)Events[DockStateChangedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockContent.cs:331]: Indentation is broken.

File: WinFormsUI/Docking/DockContentHandler.cs [hunk 4]
```diff
@@ -895,10 +895,8 @@ internal IDisposable AutoHideTab
 		}
 		protected virtual void OnDockStateChanged(EventArgs e)
 		{
-			EventHandler handler = (EventHandler)Events[DockStateChangedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler)Events[DockStateChangedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockContentHandler.cs]: Ditto

File: WinFormsUI/Docking/DockPane.cs [hunk 5]
```diff
@@ -750,10 +750,8 @@ internal void ValidateActiveContent()
 		}
 		protected virtual void OnDockStateChanged(EventArgs e)
 		{
-			EventHandler handler = (EventHandler)Events[DockStateChangedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler)Events[DockStateChangedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockPane.cs]: Here

File: WinFormsUI/Docking/DockPane.cs [hunk 6]
```diff
@@ -763,10 +761,8 @@ protected virtual void OnDockStateChanged(EventArgs e)
 		}
 		protected virtual void OnIsActivatedChanged(EventArgs e)
 		{
-			EventHandler handler = (EventHandler)Events[IsActivatedChangedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler)Events[IsActivatedChangedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockPane.cs]: there

File: WinFormsUI/Docking/DockPane.cs [hunk 7]
```diff
@@ -776,10 +772,8 @@ protected virtual void OnIsActivatedChanged(EventArgs e)
 		}
 		protected virtual void OnIsActiveDocumentPaneChanged(EventArgs e)
 		{
-			EventHandler handler = (EventHandler)Events[IsActiveDocumentPaneChangedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler)Events[IsActiveDocumentPaneChangedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockPane.cs]: etc

File: WinFormsUI/Docking/DockPanel.cs [hunk 8]
```diff
@@ -1005,10 +1005,8 @@ private bool IsClipRectsChanged(Rectangle[] clipRects)
 		}
 		protected virtual void OnContentAdded(DockContentEventArgs e)
 		{
-			EventHandler<DockContentEventArgs> handler = (EventHandler<DockContentEventArgs>)Events[ContentAddedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler<DockContentEventArgs>)Events[ContentAddedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockPanel.cs]: Here

File: WinFormsUI/Docking/DockPanel.cs [hunk 9]
```diff
@@ -1020,9 +1018,7 @@ protected virtual void OnContentAdded(DockContentEventArgs e)
 		}
 		protected virtual void OnContentRemoved(DockContentEventArgs e)
 		{
-			EventHandler<DockContentEventArgs> handler = (EventHandler<DockContentEventArgs>)Events[ContentRemovedEvent];
-			if (handler != null)
-				handler(this, e);
-		}
+            ((EventHandler<DockContentEventArgs>)Events[ContentRemovedEvent])?.Invoke(this, e);
```
Comment (HBelusca) [inline WinFormsUI/Docking/DockPanel.cs]: & there.

File: README.md (hunk 2 above)
Comment (dz333n) [reply README.md]: Converted back to 2010.

Comment (HBelusca) [reply README.md]: Does this PR indeed compile fine (with your code changes) with VS 2010?

Comment (dz333n) [reply README.md]: 🤔 I don't know, but it should. I have no VS 2010 to test.

**Comment by dz333n** [comment]
Removed 'bad' commit.

**Comment by dz333n** [comment]
[Build status](https://ci.appveyor.com/project/feel-the-dz3n/reactosdbg/branch/improvements-time)

**Comment by dz333n** [comment]
Tried to build with AppVeyor:
Visual Studio 2013: :x: [(info)](https://ci.appveyor.com/project/feel-the-dz3n/reactosdbg/builds/26618450)
Visual Studio 2015: ✅

**Comment by binarymaster** [comment]
> Conflicting files
> `RosDBG/MainWindow.Designer.cs`

@feel-the-dz3n please resolve :)

**Comment by dz333n** [comment]
I have no ability to check this after fresh commits. Conflicts are fixed, but not tested.

**Pull request closed without merging**